packages/typespec-vscode/src/vscode-cmd/create-tsp-project.ts

```diff
@@ -300,146 +325,114 @@ async function selectEmitters(
   return Object.fromEntries(selectedEmitters.map((x) => [x.name, x.emitterTemplate]));
 }
 
-async function tspInstall(
-  client: TspLanguageClient,
-  directory: string,
-): Promise<ExecOutput | undefined> {
-  logger.info("Installing TypeSpec project dependencies by 'tsp install'...");
-  return await vscode.window.withProgress(
-    {
-      location: vscode.ProgressLocation.Notification,
-      title: "Installing TypeSpec project dependencies by 'tsp install'...",
-      cancellable: true,
-    },
-    async (_progress, token) => {
-      const TIMEOUT = 600000; // set timeout to 10 minutes which should be enough for tsp install for a new project
-      try {
-        const result = await createPromiseWithCancelAndTimeout(
-          client.runCliCommand(["install"], directory),
-          token,
-          TIMEOUT,
-        );
-        return result;
-      } catch (e) {
-        if (e === ResultCode.Cancelled) {
-          logger.info(
-            "Installation of TypeSpec project dependencies by 'tsp install' is cancelled by user",
-          );
-          return undefined;
-        } else if (e === ResultCode.Timeout) {
-          logger.error(
-            `Installation of TypeSpec project dependencies by 'tsp install' is timeout after ${TIMEOUT}ms`,
-          );
-          return undefined;
+async function installDependencies(directory: string): Promise<Result<ExecOutput | undefined>> {
+  logger.info("Installing project dependencies");
+  const checkTspCliPromise = checkInstalledTspCli();
+  const checkNpmPromise = checkInstalledNpm();
+  const [checkTspCli, checkNpm] = await Promise.all([checkTspCliPromise, checkNpmPromise]);
+  if (checkTspCli || checkNpm) {
+    return await tryExecuteWithUi(
+      {
+        name: "Installing project dependencies",
+        progress: {
+          timeoutInMs: 5 * 60 * 1000, // 5 minutes
+          title: "Installing project dependencies",
+          withCancelAndTimeout: true,
+        },
+      },
+      async (progress) => {
+        if (checkTspCli) {
+          logger.info("tsp cli is installed, try to install dependencies with tsp install");
+          progress?.report({ message: "running 'tsp install' ..." });
+          return await spawnExecutionAndLogToOutput("tsp", ["install"], directory);
         } else {
-          logger.error(
-            "Unexpected error when installing TypeSpec project dependencies by 'tsp install'",
-            [e],
-          );
-          return undefined;
+          logger.info("npm is installed, try to install dependencies with 'npm install'");
+          progress?.report({ message: "running 'npm install' ..." });
+          return await spawnExecutionAndLogToOutput("npm", ["install"], directory);
         }
-      }
-    },
-  );
+      },
+    );
+  } else {
+    logger.warning(
```

Comment:
Also a follow up issue but at this point we probably should also bundle teh installDepednencies script

Reply:
Not quite sure what you mean by the "installDependencies script". Could you provide more details?

Reply:
sorry meant `installDependencies` function in compiler/src/install/install.ts` but I thikn that will need further refactor to get the cli host out of there

Reply:
yeah, it still needs some work there. I doubt the Diagnostic related function will cause some trouble too. We will work on it when working on #6321. thx.